src/prettyprinting.jl

```diff
@@ -31,46 +31,35 @@ macro objectnames(categoryname, names...)
     if Meta.isexpr(categoryname, :(=), 2)
         name = categoryname.args[1]
         category = categoryname.args[2]
-        name_str = string(name)
-        constex = if __module__ == CategoryData
-            quote
-                const $name = $category
-                function Base.show(io::IO, ::MIME"text/plain", ::Type{$name})
-                    return print(io, $name_str)
-                end
-            end
-        else
-            quote
-                const $name = $category
-                function Base.show(io::IO, ::Type{$name})
-                    return print(io, $name_str)
-                end
-                function Base.show(io::IO, ::MIME"text/plain", ::Type{$name})
-                    return print(io, $name_str)
-                end
-            end
-        end
+        constex = :(const $name = $category)
     else
         name = categoryname
         category = categoryname
         constex = :()
     end
 
+    length(names) == rank(@eval $category) ||
+        throw(ArgumentError("Number of names does not match number of objects."))
```

Comment:
```suggestion
    try
        R = rank(getglobal(TensorKitSectors, category))
        length(names) == R || throw(ArgumentError("Number of names does not match number of objects"))
    catch ex
        if ex isa UndefVarError
            throw(ArgumentError("Unknown category $category"))
        else
            rethrow(ex)
        end
    end
```

Could you try if this works instead?
I think it's slightly nicer to avoid using `@eval` here, and this gives a better error message when used incorrectly.

Reply:
I ended up doing something else, because `rank(eval(category))` doesn't check whether the category exists. Now I explicitly check it in a way that made the try-catch block redundant.